Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -255,12 +257,31 @@ internal static SFSessionProperties parseConnectionString(String connectionStrin
             checkSessionProperties(properties);
             ValidateFileTransferMaxBytesInMemoryProperty(properties);
             ValidateAccountDomain(properties);
+
+            var allowUnderscoresInHost = bool.Parse(SFSessionProperty.ALLOWUNDERSCORESINHOST.GetAttribute<SFSessionPropertyAttr>().defaultValue);

```

Comment:
I would prefer to extract this retrieving of the flag value into a separate method. And then use here something like 
```
allowUnderscoresInHost = ParseAllowUnderscoresInHost(properties)
```

because this method parseConnectionString is about 150 lines long. I think we should not continue this style.

Reply:
Done